vendor/Makefile

```
@@ -0,0 +1,152 @@
+	mkdir -p $(VENDOR_DIR)/codemirror/dist
+
+$(VENDOR_DIR)/codemirror/dist/codemirror.min.js: $(VENDOR_DIR)/codemirror/dist
+	[ -f $(VENDOR_DIR)/codemirror/dist/codemirror.min.js ] && : || \
+	curl -sSfLk --get https://cdnjs.cloudflare.com/ajax/libs/codemirror/$(CODEMIRROR_VERSION)/codemirror.min.js \
+		-z $(VENDOR_DIR)/codemirror/dist/codemirror.min.js \
+		-o $(VENDOR_DIR)/codemirror/dist/codemirror.min.js
+
+$(VENDOR_DIR)/jsplumb: vendor_dir
+	[ -d $(VENDOR_DIR)/jsplumb ] && : || \
+	git clone --depth 1 --single-branch -b $(JSPLUMB_VERSION) \
+		https://github.com/sporritt/jsPlumb.git $(VENDOR_DIR)/jsplumb
+
+$(VENDOR_DIR)/marked: vendor_dir
+	[ -d $(VENDOR_DIR)/marked ] && : || \
+	git clone --depth 1 --single-branch -b $(MARKED_VERSION) \
+		https://github.com/chjj/marked.git $(VENDOR_DIR)/marked
+
+$(VENDOR_DIR)/jquery: vendor_dir
+	mkdir -p $(VENDOR_DIR)/jquery
+
+$(VENDOR_DIR)/jquery/jquery-$(JQUERY_VERSION)-min.js: $(VENDOR_DIR)/jquery
+	[ -d $(VENDOR_DIR)/jquery/jquery-$(JQUERY_VERSION)-min.js ] && : || \
+	curl -sSfLk --get https://cdnjs.cloudflare.com/ajax/libs/jquery/$(JQUERY_VERSION)/jquery.min.js \
+		-z $(VENDOR_DIR)/jquery/jquery-$(JQUERY_VERSION)-min.js \
+		-o $(VENDOR_DIR)/jquery/jquery-$(JQUERY_VERSION)-min.js
+
+$(VENDOR_DIR)/dojotoolkit: vendor_dir
+	[ -d $(VENDOR_DIR)/dojotoolkit ] && : || (\
+	rm -rf $(VENDOR_DIR)/dojo-release-$(DOJO_VERSION).tar.gz && \
+	curl -sSfLk --get https://download.dojotoolkit.org/release-$(DOJO_VERSION)/dojo-release-$(DOJO_VERSION).tar.gz \
+		-z $(VENDOR_DIR)/dojo-release-$(DOJO_VERSION).tar.gz \
+		-o $(VENDOR_DIR)/dojo-release-$(DOJO_VERSION).tar.gz && \
+	mkdir -p $(VENDOR_DIR)/dojotoolkit && \
+	tar -xzf $(VENDOR_DIR)/dojo-release-$(DOJO_VERSION).tar.gz \
+		-C $(VENDOR_DIR)/dojotoolkit && \
+	rm -f $(VENDOR_DIR)/dojo-release-$(DOJO_VERSION).tar.gz)
+
+$(VENDOR_DIR)/highlightjs: vendor_dir
+	mkdir -p $(VENDOR_DIR)/highlightjs
+
+$(VENDOR_DIR)/highlightjs/highlight.min.js: $(VENDOR_DIR)/highlightjs
+	[ -f $(VENDOR_DIR)/highlightjs/highlight.min.js ] && : || \
+	curl -sSfLk --get http://cdnjs.cloudflare.com/ajax/libs/highlight.js/$(HIGHLIGHT_VERSION)/highlight.min.js \
+		-z $(VENDOR_DIR)/highlightjs/highlight.min.js \
+		-o $(VENDOR_DIR)/highlightjs/highlight.min.js
+
+$(VENDOR_DIR)/highlightjs/css: $(VENDOR_DIR)/highlightjs
+	mkdir -p $(VENDOR_DIR)/highlightjs/css
+
+$(VENDOR_DIR)/highlightjs/css/xcode.min.css: $(VENDOR_DIR)/highlightjs/css
+	[ -f $(VENDOR_DIR)/highlightjs/css/xcode.min.css ] && : || \
+	curl -sSfLk --get http://cdnjs.cloudflare.com/ajax/libs/highlight.js/$(HIGHLIGHT_VERSION)/styles/xcode.min.css \
+		-z $(VENDOR_DIR)/highlightjs/css/xcode.min.css \
+		-o $(VENDOR_DIR)/highlightjs/css/xcode.min.css
+
+build: build_thrift
+build: build_codemirror
+build: build_highlightjs
+build: build_marked
+build: build_jsplumb
+build: build_jquery
+build: build_dojo
+
+build_dir:
+	mkdir -p $(BUILD_DIR)
+
+build_thrift: build_dir $(VENDOR_DIR)/thrift
```

```suggestion
build_thriftjs: build_dir $(VENDOR_DIR)/thrift
```
There are multiple thrift parts in the other makefiles where the actual thrift compiler is called. I would highlight that this  is just getting the js file for thrift.